Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/EllipseLineDataProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Utilities.Lines.DataProviders
+{
+    /// <summary>
+    /// Creates an elliptical line shape.
+    /// </summary>
+    /// <remarks>This line loops.</remarks>
+    public class EllipseLineDataProvider : BaseMixedRealityLineDataProvider
+    {
+        [SerializeField]
+        private int resolution = 36;
+
+        public int Resolution
+        {
+            get { return resolution; }
+            set
+            {
+                if (value < 0)
+                {
+                    resolution = 0;
+                }
+                else if (value > 2048)
+                {
+                    resolution = 2048;
+                }
+                else
+                {
+                    resolution = value;
+                }
+            }
+        }
+
+        [SerializeField]
+        private Vector2 radius = Vector2.one;
+
+        public Vector2 Radius
+        {
+            get { return radius; }
+            set
+            {
+                if (value.x < 0)
```

Comment:
Clamp instead of the if/else

Reply:
Agreed, should use standard math functions for min and max values

Reply:
This has an unbounded max range and adding a range slider would not be ideal.